### PR TITLE
Add perfTop github CI on main branch

### PR DESCRIPTION
**Pull request by hxiao608**
Signed-off-by: Harold Xiao haoruxia@amazon.com

**Is your feature request related to a problem? Please provide an existing Issue # , or describe.**
Create CI workflow for perfTop

### Check List
- [x] New functionality includes testing.
    - [x] All tests pass
- [ ] New functionality has been documented.
    - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/perftop/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by dblock** [comment]
I see `> Task :test SKIPPED` in the log, did tests not run?

File: build.gradle [hunk 1]
```diff
@@ -51,7 +53,7 @@ node {
 }
 
 ext {
-    opendistroVersion = '1.13.0'
+    opendistroVersion = '1.0.0'
```
Comment (sruti1312) [inline build.gradle:56]: Can we rename this to ```openSearchVersion```?

**Review by sruti1312 — APPROVED** [review]

File: .github/workflows/gradle.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+          repository: 'opensearch-project/OpenSearch'
+          path: OpenSearch
+
+      - name: Build OpenSearch
+        working-directory: ./OpenSearch
+        run: ./gradlew publishToMavenLocal
+
+      - name: Run test
+        run: |
+          ./gradlew test
```
Comment (sruti1312) [inline .github/workflows/gradle.yml:42]: ```./gradlew test -Dbuild.macos=true```

File: build.gradle [hunk 3]
```diff
@@ -51,14 +53,15 @@ node {
 }
 
 ext {
-    opendistroVersion = '1.13.0'
+    opensearchVersion = '1.0.0'
```
Comment (jotok) [inline build.gradle:56]: This looks redundant with the `opensearch_version` variable declared above.

**Review by jotok — APPROVED** [review]

File: .github/workflows/gradle.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+      - name: Run test
+        run: |
+          ./gradlew test -Dbuild.linux=true
+          ./gradlew test -Dbuild.macos=true
+
+      - name: Build perfTop artifacts
+        run: |
+          ./gradlew clean
+          ./gradlew build -Dbuild.snapshot=false -Dbuild.linux=true
```
[the remark this answers is not in the document]
Comment (hxiao608) [reply .github/workflows/gradle.yml:44]: Removed the test task, and added TODO marker for future test cases

**Pull request merged**